preprocessing/create_fashion_gen_metadata.ipynb

```diff
@@ -0,0 +1,1288 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "create_fashion_gen_metadata.ipynb",
+      "provenance": [
+        {
+          "file_id": "1VOKX1mS64O4Vta7ErbYC42F9_h1alsQj",
+          "timestamp": 1631030489791
+        },
+        {
+          "file_id": "1oPsP9yM-7ZXm77MSzL9QLqmLZeZEM9Qk",
+          "timestamp": 1630618854476
+        }
+      ],
+      "last_runtime": {
+        "build_target": "//learning/deepmind/public/tools/ml_python:ml_notebook",
+        "kind": "private"
+      },
+      "toc_visible": true
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "language_info": {
+      "name": "python"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ugsdXWxylL6Z"
+      },
+      "source": [
+        "# Preprocess: Create FashionGen metadata for Mmt."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "YyB9eb8LJGiD"
+      },
+      "source": [
+        "# Defines paths and loads raw csv metadata files from Fashion-BERT and kaleido-BERT."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Cj-wbnjxMR5l",
+        "executionInfo": {
+          "status": "ok",
+          "timestamp": 1631032928352,
+          "user_tz": 420,
+          "elapsed": 2782,
+          "user": {
+            "displayName": "Roy Lu",
+            "photoUrl": "https://lh3.googleusercontent.com/a-/AOh14GjPBTCq2lmgSroOo628-irTiF0Qjk7sNdMoMMQW=s64",
+            "userId": "08276609129483219616"
+          }
+        }
+      },
+      "source": [
+        "import pandas as pd\n",
+        "import tensorflow as tf\n",
+        "\n",
+        "# i2t: image-to-text.\n",
+        "i2t_path = '/bigstore/mmt/raw_data/fashion_gen/fashion_gen_i2t_test_pairs.csv'\n",
+        "# t2i: text-to-image.\n",
+        "t2i_path = '/bigstore/mmt/raw_data/fashion_gen/fashion_gen_t2i_test_pairs.csv'\n",
+        "\n",
+        "t2i_output_path = '/bigstore/mmt/fashion_gen/metadata/fashion_bert_t2i_test.csv'\n",
+        "i2t_output_path = '/bigstore/mmt/fashion_gen/metadata/fashion_bert_i2t_test.csv'\n",
+        "\n",
+        "dtype = {\n",
+        "    'image_prod_id': str,\n",
+        "    'prod_img_id': str,\n",
+        "    'text_prod_id': str,\n",
+        "}\n",
+        "\n",
+        "with tf.io.gfile.GFile(i2t_path, 'r') as f:\n",
+        "  i2t_df = pd.read_csv(f, dtype=dtype)\n",
+        "with tf.io.gfile.GFile(t2i_path, 'r') as f:\n",
+        "  t2i_df = pd.read_csv(f, dtype=dtype)"
+      ],
+      "execution_count": 39,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "height": 419
+        },
+        "id": "YMCqOYW9__M9",
+        "executionInfo": {
```

Comment:
Are you sure you want this type of info to be checked in? there is even a link to a picture of you here.

Reply:
And same for several other blocks below.

Reply:
+1 it's hard to review this file:)

Reply:
@santiaontanon-google @wqfcr can you use the following link to review? It's the same file.
https://github.com/googleinterns/multimodal-long-transformer-2021/blob/ae21f61e1f69beeba82b7bb0ea02087bc90c02ea/preprocessing/create_fashion_gen_metadata.ipynb